lms/djangoapps/django_comment_client/base/tests.py

```diff
@@ -1095,6 +1104,221 @@ def _test_unicode_data(self, text, mock_request):
             del Thread.commentable_id
 
 
+@ddt.ddt
+@patch("lms.lib.comment_client.utils.requests.request")
+class TeamsPermissionsTestCase(UrlResetMixin, ModuleStoreTestCase, MockRequestSetupMixin):
+    # Most of the test points use the same ddt data.
+    # args: user, commentable_id, status_code
+    ddt_args = [
```

Comment:
This is excellent! I love how many combinations are being tested here.

Nit: I'd rename this to something that describes what it is, e.g. ``ddt_permission_cases`` or something.